frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts

```diff
@@ -181,41 +195,63 @@ function isUserActivity(snapshot: eventWithTime): boolean {
 
 const updatePlayerTimeTracking = (
     current: PlayerTimeTracking,
-    newState: PlayerTimeTracking['state']
+    newState: PlayerTimeTracking['state'],
+    openTime?: number
 ): PlayerTimeTracking => {
+    const now = performance.now()
+
     // if we were just playing then update watch time
     const newWatchTime =
         current.lastTimestamp !== null && current.state === 'playing'
-            ? current.watchTime + (performance.now() - current.lastTimestamp)
+            ? current.watchTime + (now - current.lastTimestamp)
             : current.watchTime
 
     // if we were just buffering then update buffer time
     const newBufferTime =
         current.lastTimestamp !== null && current.state === 'buffering'
-            ? current.bufferTime + (performance.now() - current.lastTimestamp)
+            ? current.bufferTime + (now - current.lastTimestamp)
             : current.bufferTime
 
-    const newLastTimestamp = ['paused', 'ended', 'errored'].includes(newState) ? null : performance.now()
+    const newLastTimestamp = ['paused', 'ended', 'errored'].includes(newState) ? null : now
+
+    let newFirstPlayStartTime = current.firstPlayStartTime
+    let newFirstPlayTime = current.firstPlayTime
+
+    if (newState === 'playing' && current.firstPlayStartTime === undefined && current.firstPlayTime === undefined) {
+        newFirstPlayStartTime = now
+    } else if (['paused', 'ended', 'errored'].includes(newState)) {
+        newFirstPlayStartTime = undefined
+    } else if (
+        newState === 'buffering' &&
+        current.firstPlayTime === undefined &&
+        current.firstPlayStartTime !== undefined
+    ) {
+        // If we go back to buffering before completing the 1-second threshold, reset tracking
+        if (now - current.firstPlayStartTime < 1000) {
+            newFirstPlayStartTime = undefined
+        }
+    }
+
+    if (
+        current.firstPlayTime === undefined &&
+        current.firstPlayStartTime !== undefined &&
+        newState === 'playing' &&
+        now - current.firstPlayStartTime >= 1000 &&
+        openTime !== undefined
+    ) {
+        newFirstPlayTime = current.firstPlayStartTime - openTime
+    }
```

Comment:
It's very tricky to follow what is going on through all these conditionals. Maybe a comment explaining what the difference between firstPlayTime and firstPlayStartTime is and how this method aims to update them?

Or maybe there is a way to pull some of those many comparisons out into separate conditionals?